libs/build_esmf.sh

```diff
@@ -110,6 +110,7 @@ case $MPI in
     export ESMF_COMM=${STACK_esmf_comm:-"mpich3"}
     ;;
   cray-mpich )
+    export ESMF_OS="Linux"
```

Comment:
@aerorahul @mark-a-potts Still not working. I think this line ESMF_OS="Linux" is the culprit. In the previous hpc stack v1.1.0 build log I see that it autodetecttedd ESMF_OS="Unicos" for Cray, as it should be.

Reply:
@climbfuji @mark-a-potts 
I am removing this line from this PR and placing it in `config_wcoss2.sh` for posterity.